### ISSUE TITLE
Enhance error output from failed `get` functions.

### ISSUE THREAD

**Issue by phillipedwards** [pulumi/pulumi#10655]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Currently, when a `get` function such as `aws.ec2.getSecurityGroup` fails or errors, the response is very generic and provides little insight into the scenario in which the error occurred. 

Example of current error:
```
error: an unhandled error occurred: 1 error occurred:
    	* rpc error: code = Unknown desc = invocation of aws:ec2/getSecurityGroup:getSecurityGroup returned an error: invoking aws:ec2/getSecurityGroup:getSecurityGroup: 1 error occurred:
    	* no matching SecurityGroup found
```

This might be ok for small programs or programs where the inputs to the `get` function are static, however, it is difficult to debug these errors in large programs with multiple `get` invocations of the same type and/or the program is driven by a large configuration (JSON, YAML, etc) dataset (eg- a reusable function used in connection with a large configuration dataset).

### Affected area/feature

To enhance the error output, I'd like to set the line of the user program in which the error was realized and also the inputs provided to the `get` function which precipitated the error. Eg- 

```
const sg = await aws.ec2.getSecurityGroup({id: "my-id");
```

If the above example were to fail, I'd like to see something along the lines of:
```
error: an unhandled error occurred: 1 error occurred **on line 23**:
    	* rpc error: code = Unknown desc = invocation of aws:ec2/getSecurityGroup:getSecurityGroup returned an error: invoking aws:ec2/getSecurityGroup:getSecurityGroup: 1 error occurred:
    	* no matching SecurityGroup found **for inputs '{id: "my-id"}**'
```

**Comment by iwahbe** [comment]
I don't think we are able to support line numbers from the RPC layer. We could `throw`/`raise` errors in supporting languages so line numbers become accessible. @dixler are you interested in picking this up as part of the "first 48H" set of issues?

**Comment by joeduffy** [comment]
I wonder, is there any reason we can’t capture a stack trace at the time of invocation and use that in the error? IIRC we already do capture it but only use it in debug tracing and not the error itself.
